src/foam/nanos/dig/DIG.js

```diff
@@ -31,68 +36,80 @@ foam.CLASS({
     }
   ],
 
+  imports: [
+    'nSpecDAO'
+  ],
+
   properties: [
-    'id',
     {
-      class: 'String',
+      name: 'id',
+      label: 'Request Name'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.boot.NSpec',
+      label: 'Data Access Object (DAO)',
       name: 'daoKey',
-      label: 'DAO',
+      documentation: `The DAO in the DIG request.`,
       view: function(_, X) {
         var E = foam.mlang.Expressions.create();
-        return foam.u2.view.ChoiceView.create({
-          dao: X.nSpecDAO
-            .where(E.ENDS_WITH(foam.nanos.boot.NSpec.ID, 'DAO'))
-            .where(E.EQ(foam.nanos.boot.NSpec.SERVE, E.TRUE))
-            .orderBy(foam.nanos.boot.NSpec.ID),
-          objToChoice: function(nspec) {
-            return [nspec.id, nspec.id];
-          }
-        });
+        return {
+          class: 'foam.u2.view.RichChoiceView',
+          search: true,
+          sections: [
+            {
+              heading: 'DAO',
+              dao: X.nSpecDAO
+                .where(E.AND(
+                  E.EQ(foam.nanos.boot.NSpec.SERVE, true),
+                  E.ENDS_WITH(foam.nanos.boot.NSpec.ID, 'DAO')
+                ))
+                .orderBy(foam.nanos.boot.NSpec.ID)
+            }
+          ]
+        };
       },
-      value: 'accountDAO'
+      value: 'userDAO'
     },
-    'cmd',
-    'format',
     {
-      class: 'String',
-      name: 'dao',
-      hidden: true,
-      transient: true,
-      postSet: function(old, nu) {
-        this.daoKey = nu;
+      name: 'cmd',
+      label: 'API Command'
+    },
+    {
+      name: 'format',
+      label: 'Data Format',
+      visibility: function(cmd) {
+        return ( cmd == 'SELECT' || cmd == 'PUT' ) ? foam.u2.DisplayMode.RW : foam.u2.DisplayMode.HIDDEN;
       }
     },
     {
+      class: 'String',
+      name: 'key',
+      label: 'Object ID',
+      visibility: function(cmd) {
+        return ( cmd == 'SELECT' || cmd == 'REMOVE' ) ? foam.u2.DisplayMode.RW : foam.u2.DisplayMode.HIDDEN;
+      }
+  },
+  {
       class: 'String',
       name: 'q',
-      label: 'Query'
+      label: 'Select Query',
+      visibility: function(cmd) {
+        return (cmd == 'SELECT') ? foam.u2.DisplayMode.RW : foam.u2.DisplayMode.HIDDEN;
```

Comment:
Should be visible for 'REMOVE" also, shouldn't it?

Reply:
Or do we only accept remove by primary key? (Which is less useful, but safer.)

Reply:
yeah .. we require a PK.